### ISSUE TITLE
Running assume-role resets PWD and bash_history

### ISSUE THREAD

**Issue by osulli** [cloudposse/geodesic#400]
## what

- Running `assume-role` resets the PWD (present working directory)
- Running `assume-role` resets the bash history
## why

- This is a problem because it means upon changing AWS **security levels** the entire shell is basically reset. I cannot think of a use case where I am escalating privileges but want to lose my history and PWD.

### My OS

```
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.10
Release:	18.10
Codename:	cosmic
```
### Geodesic

```
FROM cloudposse/terraform-root-modules:0.53.4 as terraform-root-modules

FROM cloudposse/helmfiles:0.19.1 as helmfiles

FROM cloudposse/geodesic:0.73.0
```

**Comment by osterman** [comment]
It's not resetting history, so much as `aws-vault` is starting a new shell.

**Comment by osterman** [comment]
History is preserved across shell sessions. Hit `^R` and you'll see what I mean.

**Comment by osulli** [comment]
I don't thikn History is preserved @osterman, I would have noticed if it was because Reverse Search is exactly what I wanted to do.

```
 ⧉  mgmt 
 ✓   (healthera-mgmt-admin) iam_roles ⨠ history
    1  cd /localhost/code/devops/terraform/providers/aws/iam_roles/
    2  terraform apply
    3  history
    4  terraform apply
    5  clear
    6  history
 ⧉  mgmt 
 ✓   (healthera-mgmt-admin) iam_roles ⨠ echo "test"
test
 ⧉  mgmt 
 ✓   (healthera-mgmt-admin) iam_roles ⨠ history
    1  cd /localhost/code/devops/terraform/providers/aws/iam_roles/
    2  terraform apply
    3  history
    4  terraform apply
    5  clear
    6  history
    7  echo "test"
    8  history
 ⧉  mgmt 
 ✓   (healthera-mgmt-admin) iam_roles ⨠ logout
Goodbye
-> Run 'assume-role' to login to AWS with aws-vault
⧉  mgmt 
 ✗   (none) ~ ⨠ assume-role
Enter passphrase to unlock /conf/.awsvault/keys/: 
* Assumed role arn:aws:iam::065320943774:role/MgmtAccessFromSecurityAcc
 ⧉  mgmt 
 ✓   (healthera-mgmt-iac) ~ ⨠ history
    1  history
 ⧉  mgmt 
(failed reverse-i-search)`echo': 
```

**Comment by osterman** [comment]
[here is where we set the history file](https://github.com/cloudposse/geodesic/blob/77888c94e10419be60b3adafd18527cfa1d83d95/Dockerfile#L210). Maybe it's related to filesystem permissions?

**Comment by osulli** [comment]
@osterman I'd like to help more on this, as I'm still plagued by it and likely other Ubuntu users are.

The URL you sent me didn't show me what my next step was. Upon looking at it a week later, I think we're actually talking about different things.

Your link was to the Dockerfile and your caching scenario is when you rebuild the Docker image. Haven't confirmed this myself.
My issue's scenario is I run `assume-role` inside the shell and my whole bash history is wiped.

**Comment by osterman** [comment]
Please have a look at how [bash history](https://www.gnu.org/software/bash/manual/html_node/Bash-History-Facilities.html) works. I linked you to where we set the envs related to the bash history. We're not resetting anything. Not trying to minimize your problem, but I cannot reproduce it.  =) My history is preserved.

**Comment by osulli** [comment]
Fixed on geodesic:0.77.2 from 0.73.0

**Comment by osterman** [comment]
@osulli tons of bug fixes/enhancements by @Nuru in this PR: https://github.com/cloudposse/geodesic/pull/422
